service/history/shardController.go

```diff
@@ -162,8 +162,11 @@ func (c *shardController) Stop() {
 }
 
 func (c *shardController) GetEngine(workflowID string) (Engine, error) {
-	shardID := common.WorkflowIDToHistoryShard(workflowID, c.config.NumberOfShards)
-	return c.getEngineForShard(shardID)
+	return c.getEngineForShard(c.GetShardID(workflowID))
+}
+
+func (c *shardController) GetShardID(workflowID string) int {
```

Comment:
It looks like an overkill for now.  Number of shards cannot be changed for a cluster.  Just define a helper method rather than expose this from ShardController.

Reply:
sure